lib/spack/spack/solver/asp.py

```diff
@@ -2889,26 +2910,45 @@ def depends_on(self, dependency_str: str, *, when: str, type: str, description:
             f"  not external({node_variable}),\n"
             f"  not runtime(Package)"
         ).replace(f'"{placeholder}"', f"{node_variable}")
+        if languages:
+            body_str += ",\n"
+            for language in languages:
+                body_str += f'  attr("language", {node_variable}, "{language}")'
```

Comment:
this, like `depends_on()` should be using `SUPPORTED_LANGUAGES` an not forcing extra parameters on the caller. 

TODO: the final implementation of this should not have an extra `languages` parameter and should not have any special cases for "languages".  There should be one general mechanism for packages to inject additional dependencies, and that's all we should need.